### DISCUSSION TITLE
Collage-Settings for two small 2x6" - Photo-Stripes

### DISCUSSION THREAD

**Discussion by msmedien** [photobooth-app/photobooth-app#344] [Q&A]
Hello,

I tried to use the collage function to create a template that would be cut into two narrow 2x6" strips of images at the end. I can set the printer (DNP) to do this without any problems and it cuts the image into both strips. - BUT I can't manage the fact that the image taken is duplicated on both sides of the strips. I can define 6 photos and their position, but then I have 6 different photos... But I want the guests to take 3 photos and then have them in 6 positions on the final collage. (see example image)

Can anyone give me a tip here? I always set up the collage image at 4x6" and then had it cut. Or do I have to think the other way around and set up the collage at 2x6" and only print out individual strips? - If the printer even does that...

![Stripes](https://github.com/user-attachments/assets/c1f6174f-8b12-43a8-8665-0a6cc6dcae97)

**Comment by mgineer85** [comment]
Hi, currently there is no way to have two equal captures in one collage. 
Simplest workaround I could think of is to create just the left / right side and print two pages of that at once. Would that work for you?

**Reply by msmedien** [reply]
Hello :)

Sadly, your suggestion doesn't work. The printer always needs the basic format of 4x6". It can then either print this format over the entire surface or cut it into two 2x6" strips.

Is there a way to simply duplicate the final collage image (background plus the 3 photos) and place it next to each other twice?

I could imagine the function as follows: In the options for the collage, a strip of 2x6" (e.g. 800x2400px) is created, with 3 photos and then there would be an option to click "Duplicate collage for strips". The output of the final collage would then be a 4x6" (1600x2400px) image.

**Reply by mgineer85** [reply]
I think this is quite individual to a specific setup. So instead implementing it in the app I would propose to create a simple script doing that job.

See this one as a reference how it could work: https://askubuntu.com/questions/433236/how-to-print-multiple-copies-of-an-image-on-a-single-page

This is very flexible and you can tailor very specific to your printer.

When you find a solution, it would be great if you share so other can take advantage of your implementation also.

**Reply by PixelAlarm** [reply]
Hi,
this can be easily achieved with a small script.
I also use ist for Collage-Printing with a DNP-based printer.

Look here for reference:
https://github.com/photobooth-app/photobooth-app/discussions/283

**Comment by Tyfab56** [comment]
did you manage to do it ?

**Reply by msmedien** [reply]
Thanks for asking. Unfortunately, I have a lot of work to do this week and haven't had a chance to do it yet. Next week I'll finally have time for my photobooth and will try out your tips. :) I'll keep you updated.

**Comment by Kingvic** [comment]
I was able to do this with a little bit of help from ChatGPT. 
I use a 2*6 canvas within photobooth for the collage and run the following script as the print command. It auto detect whether it is a collage or a single image based on the photo ratio size.

- usage command: /home/pi/printscript.sh {filename}

- set PRINTER to the printer name within CUPS

```
#!/bin/bash
#
# Smart print script for 2x6 photo strips
# If input is 1:3 (e.g. 2x6), duplicates and adds cut line for 4x6 output
# Prints portrait for collage, landscape otherwise
# Requires: ImageMagick, CUPS
#

# === CONFIG ===
PRINTER="PRINTER1"
TMP_COLLAGE="/tmp/collage.jpg"

# === INPUT CHECK ===
PICTURE="$1"
if [ -z "$PICTURE" ] || [ ! -f "$PICTURE" ]; then
  echo "Usage: $0 {image filename}"
  exit 1
fi

# === DIMENSION + ASPECT RATIO DETECTION ===
width=$(identify -format '%w' "$PICTURE")
height=$(identify -format '%h' "$PICTURE")

# GCD to reduce ratio
gcd() {
  local a=$1
  local b=$2
  while [ "$b" -ne 0 ]; do
    local t=$b
    b=$((a % b))
    a=$t
  done
  echo "$a"
}

gcd_val=$(gcd "$width" "$height")
aspect_w=$((width / gcd_val))
aspect_h=$((height / gcd_val))

echo "Detected aspect ratio: $aspect_w:$aspect_h"

# === COLLAGE MODE: 1:3 (e.g., 2x6 strip) ===
if [[ "$aspect_w" -eq 1 && "$aspect_h" -eq 3 ]]; then
  echo "Detected 1:3 ratio — treating as collage (2x6 duplicated)"

  # Duplicate side-by-side
  convert "$PICTURE" "$PICTURE" +append "$TMP_COLLAGE"

  # Add vertical cut line in center
  collage_width=$(identify -format '%w' "$TMP_COLLAGE")
  collage_height=$(identify -format '%h' "$TMP_COLLAGE")
  mid_x=$((collage_width / 2))

  convert "$TMP_COLLAGE" -stroke black -strokewidth 3 -draw "line $mid_x,0 $mid_x,$collage_height" "$TMP_COLLAGE"

  # Print in portrait (no scaling)
  if lp -d "$PRINTER" -o orientation-requested=3 "$TMP_COLLAGE"; then
    echo "Print successful, cleaning up."
    rm -f "$TMP_COLLAGE"
  else
    echo "Print failed, not deleting temp file: $TMP_COLLAGE"
  fi

# === DEFAULT MODE: print single image in landscape ===
else
  echo "Single image — printing in landscape"
  lp -d "$PRINTER" -o orientation-requested=4 "$PICTURE"
fi

```
